frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/MfaHandler.java

```diff
@@ -249,28 +249,58 @@ private Optional<ErrorResponse> validateCodeRequestAttempts(
         var newCodeBlockPrefix = CODE_BLOCKED_KEY_PREFIX + codeRequestType;
 
         if (codeRequestCount == configurationService.getCodeMaxRetries()) {
-            LOG.info(
-                    "User has requested too many OTP codes. Setting block with prefix: {}",
-                    newCodeRequestBlockPrefix);
-            codeStorageService.saveBlockedForEmail(
-                    email, newCodeRequestBlockPrefix, configurationService.getLockoutDuration());
-            LOG.info("Resetting code request count");
-            sessionService.save(
-                    session.resetCodeRequestCount(NotificationType.MFA_SMS, journeyType));
+            LOG.warn("User has requested too many OTP codes.");
+
+            blockReauthenticatingUserWhenReauthenticationLogOffNotSupported(
+                    email, journeyType, newCodeRequestBlockPrefix);
+
+            blockUsersOnAllJourneysOtherThanReauthenticatingUsers(
+                    email, journeyType, newCodeRequestBlockPrefix);
+
+            clearCountOfFailedCodeRequests(journeyType, session);
+
             return Optional.of(ErrorResponse.ERROR_1025);
         }
+
         if (codeStorageService.isBlockedForEmail(email, newCodeRequestBlockPrefix)) {
             LOG.info(
                     "User is blocked from requesting any OTP codes. Code request block prefix: {}",
                     newCodeRequestBlockPrefix);
             return Optional.of(ErrorResponse.ERROR_1026);
         }
+
         if (codeStorageService.isBlockedForEmail(email, newCodeBlockPrefix)) {
             LOG.info(
                     "User is blocked from entering any OTP codes. Code attempt block prefix: {}",
                     newCodeBlockPrefix);
             return Optional.of(ErrorResponse.ERROR_1027);
         }
+
         return Optional.empty();
     }
+
+    private void blockReauthenticatingUserWhenReauthenticationLogOffNotSupported(
+            String email, JourneyType journeyType, String newCodeRequestBlockPrefix) {
+        if (journeyType == JourneyType.REAUTHENTICATION
+                && !configurationService.supportReauthSignoutEnabled()) {
```

Comment:
Not strictly related to this PR, but I notice we have duplicate configuration service methods reading from the same env var (`supportReauthSignoutEnabled` and `isReauthSignoutEnabled`) - I think we should raise a subsequent PR after this one to consolidate

Reply:
Removed isReauthSignoutEnabled in favour of supportReauthSignoutEnabled.